src/PrestaShopBundle/Utils/ImmutableFloat.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace PrestaShopBundle\Utils;
+
+/**
+ * Immutable value-object that contains a float value.
+ */
+class ImmutableFloat
```

Comment:
> > moreover, core classes such as \DateTime have Immutable as suffix

> So if it was FloatImmutable it would be OK?

No/Yes .. if your last decision is to reatain the Immutable world in the name I think it is just more consistent .. but my approach is the one that follows

> > you are putting something resembling a Type, a VO, a Class into the "Utils" namespace

> I agree that Utils may not be the best place to put it. What do you suggest?

Neglet the \Numberformatter part, something as exposed here (make the class name an "ActionMaker", and utility .. not a "type"): https://github.com/PrestaShop/PrestaShop/pull/7934#pullrequestreview-42821140